expan/core/experiment.py

```diff
@@ -287,11 +287,64 @@ def _quantile_filtering(self, data, kpis, percentile, threshold_type):
             flags = flags | data[column].apply(method_table[threshold_type])
         return flags
+
+
+    def chi_square_test_result_and_statistics(self, variant_column, weights, min_counts=5, alpha=0.05):
+        """ Tests the consistency of variant split with the hypothesized distribution.
+        
+        :param variant_column: variant column from the input data frame
+        :param weights: dict with variant names as keys, weights as values
+                        ({<variant_name>:<weight>, ...}
+        :param min_counts: minimum number of observed and expected frequencies (should be at least 5), see 
+                            http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
+        :param alpha: significance level, 0.05 by default
+        :return: True(if split is consistent with the given split) or
+                 False(if split is not consistent with the given split)
+        :rtype:  Boolean, float, float
+        """
+        if not variant_column:
+            raise ValueError("Variant split check was cancelled since input variant column is empty or doesn't exist.")
+        if not weights:
+            raise ValueError("Variant split check was cancelled since input weights are empty or doesn't exist.")
+        if len(weights) <= 1 or len(variant_column) <= 1:
+            raise ValueError("Variant split check was cancelled since input weights or the number if categories "
+                             "is less than 2.")
+
+        # Count number of observations per each variant
+        variant_column = pd.Series(variant_column).dropna(axis=0)
+        variants_counts = variant_column.value_counts()
+        observed_counts = pd.DataFrame(variants_counts, columns=['observed'])
+        # Ensure at least a frequency of 5 at every location in observed_counts, otherwise drop the category
+        # see http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
+        observed_freqs = observed_counts[observed_counts >= min_counts]
+
+        # If all categories were dropped or if there are less than 2 categories left we can't conduct the test.
+        if observed_freqs.empty or len(observed_freqs) < 2:
+            raise ValueError("Category frequencies are empty or number of categories is less than 2. "
+                             "Chi-square test is not applicable.")
+
+        # Calculate expected counts given corresponding weights
+        total_count = observed_freqs.sum().sum()
+        expected_freqs = pd.DataFrame.from_dict(weights, orient='index')
+        expected_freqs.columns = ['weight']
+        expected_freqs *= total_count
+
+        # Join all values in one data frame, so that each observed category count corresponds to the correct expected
+        # count defined by the corresponding weight or split percentage.
+        all_freqs = expected_freqs.join(observed_freqs).dropna(axis=0)
```

Comment:
why do you join ```expected_freqs``` and ```observed_freqs``` into one dataframe ```all_freqs```?

The only use of ```all_freqs``` (I found) is to separate it back into two dataframe in the next lines: ```statx.chi_square(all_freqs['observed'], all_freqs['weight'])```

Reply:
I did that because I need the correct corresponding observed and expected count per variant, expected_freqs and observed_freqs are not sorted, so, the values can be mixed. For example, with all_freqs I will have 

   weight  observed
A    25.0        23
B    15.0        18
C    10.0        17
D    40.0        19
E    10.0        23

separately my lists will be 
A    25.0
B    15.0
C    10.0
D    40.0
E    10.0
 and 

  observed
E        23
A        23
D        19
B        18
C        17
 I can sort both as a solution.

Reply:
sort_index is simpler =)